spock-specs/src/test/groovy/org/spockframework/smoke/mock/InvokingMocksFromMultipleThreads.groovy

```diff
@@ -24,9 +24,11 @@ import org.spockframework.mock.TooManyInvocationsError
 import spock.lang.*
 
 class InvokingMocksFromMultipleThreads extends Specification {
-  def numThreads = 10
+  private static final int WAIT_TIME_S = 200
+
+  def numThreads = WAIT_TIME_S
```

Comment:
Creating 200 threads could take some time on slower machines.

At least for Java 21 we could use virtual threads.

```groovy
  @AutoCleanup("shutdownNow")
  ExecutorService executorService = createExecutorService()

  ExecutorService createExecutorService() {
    Jvm.current.java21Compatible ? Executors."newVirtualThreadPerTaskExecutor"() : Executors.newCachedThreadPool {new Thread(it).tap { it.daemon = true } }
  }
```